src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.cs

```diff
@@ -18,6 +18,11 @@ public static partial class RuntimeHelpers
         /// <summary>
         /// Slices the specified array using the specified range.
         /// </summary>
+#if NET6_0_OR_GREATER
+        [System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessage("ArrayCreateInstance", "IL3050:RequiresDynamicCode",
+            Justification = @"The RequiresDynamicCode warning is about Array.CreateInstance. However, we must have
+             preserved the code to create a U : T array, if the incoming array is U : T.")]
+#endif
```

Comment:
```suggestion
```
This should not be needed anymore. I have changed the implementation to not call Array.CreateInstance recently.

Reply:
Yup, I have to integrate with main to pull that in.